rust/private/rustc.bzl

```diff
@@ -1063,6 +1063,9 @@ def construct_arguments(
     else:
         rustc_flags.add_all(toolchain.extra_rustc_flags)
 
+    if crate_info.type in ["rlib", "lib"] and not is_exec_configuration(ctx):
+        rustc_flags.add_all(toolchain.extra_rustlib_rustc_flags)
```

Comment:
I think the attribute naming is too generic whereas its responsibility is too specific (i.e. only applying to rlib or lib and not in exec configuration).

1. If we want to apply the flags to non-exec configuration, we can just set the flags to non-exec toolchains. Exec and non-exec are separate toolchains.

2. "rustlib" means more than just rlib and lib. It can include dylib (not implemented yet), cdylib as well.

I feel like the rust_tochain rule isn't the right level for this kind of flag.

Normally for scenarios like this, we could create macros to wrap the rust_library and rust_static_library rules with the custom flags instead? Would this be a solution?

Reply:
I agree with these concerns, but I think with a refactor it could be useful. How about instead of `extra_rustlib_rustc_flags = ["--foo"]`, we add `extra_rustc_flags_for_crate_types = {"rlib": ["--foo"], "lib": ["--foo"]}` (and remove the `is_exec_configuration` check). I believe that would be generic enough to be worth adding to the toolchain.